wavefront/wavefront.go

```
@@ -101,21 +119,70 @@ func (wa *WavefrontAdapter) setLogLevel(cfg *config.Params) {
 // verifyAndInitReporter checks if the Wavefront reporter is initialized, and if
 // not, initializes it.
 func (wa *WavefrontAdapter) verifyAndInitReporter(cfg *config.Params) {
-	if !wa.reporterInitialized {
+	if wa.reporter == nil {
 		log.Infof("trying to init wavefront reporter, config: %s", cfg.String())
-
 		wa.setLogLevel(cfg)
 
 		if err := config.ValidateCredentials(cfg); err != nil {
 			log.Errorf("failed to create wavefront reporter, err: %s, config: %s", err.Error(), cfg.String())
 		} else {
-			createWavefrontReporter(cfg)
-			wa.reporterInitialized = true
+			wa.createWavefrontReporter(cfg)
 			log.Infof("wavefront reporter successfully initialized, config: %s", cfg.String())
 		}
 	}
 }
 
+// creates wavefront direct sender
+func createDirectSender(direct *config.Params_WavefrontDirect, flushInterval int) senders.Sender {
+	directCfg := &senders.DirectConfiguration{
+		Server:               direct.Server,
+		Token:                direct.Token,
+		FlushIntervalSeconds: flushInterval,
+		BatchSize:            10000,
+		MaxBufferSize:        50000,
+	}
+	sender, err := senders.NewDirectSender(directCfg)
+	if err != nil {
+		log.Fatalf("Error creating direct sender: %v", err)
+		return nil
+	}
+	return sender
+}
+
+// creates wavefront proxy sender
+func createProxySender(proxy *config.Params_WavefrontProxy, flushInterval int) senders.Sender {
+	addr, _ := net.ResolveTCPAddr("tcp", proxy.Address)
```

Don't ignore the error.

---

Done.